Fixed issue related to "no documents in result" on first run

Yoone opened the pull request:
# Context

I was getting the following error when no documents were present in the `migrations` collection (first run):

```
mongo: no documents in result
```

# Fix

Adding an additional check (exact same condition as the one a few lines below) fixed that issue.

xakep666 commented:
Thanks for pull request. That's will be great if you also write test for it

xakep666 commented on migrate.go:
```diff
@@ -139,6 +139,10 @@ func (m *Migrate) Version() (uint64, string, error) {
 	// find record with greatest id (assuming it`s latest also)
 	result := m.db.Collection(m.migrationsCollection).FindOne(context.TODO(), filter, options)
 	if err := result.Err(); err != nil {
+		if err == mongo.ErrNoDocuments {
```
I think switch-case (with nil, ErrNoDocuments and default cases) here will look better

Yoone commented:
* I changed the error check to use a switch like you asked. I also removed the second `err == mongo. ErrNoDocuments` right below because it will now never reach it anymore.
* I noticed a missing error check in `TestMain`, so I added that.

In terms of tests, I only _moved_ the error check above, so there's nothing _new_ to be tested.

When I display the coverage though, the previous errors are never reached by the current tests:

![image](https://user-images.githubusercontent.com/2976044/63684723-e0949d00-c7fd-11e9-9c2b-0f592f9dabeb.png)

I'm not sure how to reach them though, as they're actual DB errors.

Yoone commented:
@xakep666 I don't think I did what you were asking for in terms of switch-case. Please tell me if this would be closer to what you were thinking:

```go
	result := m.db.Collection(m.migrationsCollection).FindOne(context.TODO(), filter, options)
	err := result.Err()
	switch err {
	case mongo.ErrNoDocuments:
		return 0, "", nil
	case nil:
		var rec versionRecord
		if err := result.Decode(&rec); err != nil {
			return 0, "", err
		}
		return rec.Version, rec.Description, nil
	default:
		return 0, "", err
	}
```

I also think it's safe to just do the following (not checking the first error, checking it after calling `Decode`).

If you look at https://github.com/mongodb/mongo-go-driver/blob/master/mongo/single_result.go#L35-L38, it returns the existing error if the field was already set. Code would look like this (it passes the tests):

```go
	var rec versionRecord
	err := m.db.Collection(m.migrationsCollection).
		FindOne(context.TODO(), filter, options).
		Decode(&rec)
	switch err {
	case mongo.ErrNoDocuments:
		return 0, "", nil
	case nil:
		return rec.Version, rec.Description, nil
	default:
		return 0, "", err
	}
```

xakep666 commented:
@Yoone your switch-case variant is better than I thought. Thank you!
